datafusion/optimizer/tests/optimizer_integration.rs

```diff
@@ -344,7 +345,8 @@ fn test_sql(sql: &str) -> Result<LogicalPlan> {
         .with_udaf(sum_udaf())
         .with_udaf(count_udaf())
         .with_udaf(avg_udaf());
-    let sql_to_rel = SqlToRel::new(&context_provider);
+    let sql_to_rel = SqlToRel::new(&context_provider)
+        .with_user_defined_planner(Arc::new(CoreFunctionPlanner::default()));
```

Comment:
We intend to avoid `optimizer` crate dependent on `functions` .
We can try to move the test to `datafusion/core/tests/optimizer_integration.rs`

Reply:
As suggested moved all the test cases which were dependent of `functions`.